posthog/hogql/ast.py

```diff
@@ -319,6 +319,19 @@ class OrderExpr(Expr):
     order: Literal["ASC", "DESC"] = "ASC"
 
 
+class ArrayAccess(Expr):
+    array: Expr
```

Comment:
Loose question: I wonder why we can't type this as `Array`?

Reply:
sometimes arrays are returned from elsewhere... e.g `arrayMap(a->a*2, [1,2,2])[1]`